Block/Service/AbstractCategoriesBlockService.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Block\Service;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\ClassificationBundle\Model\CategoryInterface;
+use Sonata\ClassificationBundle\Model\CategoryManagerInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
+use Sonata\CoreBundle\Model\Metadata;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractCategoriesBlockService extends AbstractClassificationBlockService
+{
+    /**
+     * @var CategoryManagerInterface
+     */
+    private $categoryManager;
+
+    /**
+     * @var AdminInterface
+     */
+    private $categoryAdmin;
+
+    /**
+     * @param string                   $name
+     * @param EngineInterface          $templating
+     * @param ContextManagerInterface  $contextManager
+     * @param CategoryManagerInterface $categoryManager
+     * @param AdminInterface           $categoryAdmin
+     */
+    public function __construct($name, EngineInterface $templating, ContextManagerInterface $contextManager, CategoryManagerInterface $categoryManager, AdminInterface $categoryAdmin)
```

Comment:
> This discussion was already made and closed, no need for more...

So people who are late are wrong ? The only argument that makes sense in the Sonata discussion IMO is "consistency". And it does not make sense in the Symfony discussion, because if you change long method signature to multi line, then you are consistent.

On a side note, I see more and more bad things done in Sonata for the sake of consistency, no later than this morning : "well this is how it is done in SonataXXXBundle", and in this case, if you have short signatures onelined and long signatures multilined, you can't really call that "inconsistent", can you? You can only if you have some long signatures multilined and some other long signatures onelined.